Source/Core/VideoBackends/D3D12/Render.cpp

```diff
@@ -1244,6 +1242,8 @@ void Renderer::SetSamplerState(int stage, int tex_index, bool custom_tex)
 	if (tex_index)
 		stage += 4;
 
+	const auto original_hex = gx_state.sampler[stage].hex;
```

Comment:
What do you think about instead creating a copy, just creating a new state object and apply it if it is different than the current bound one?